src/Symfony/Bundle/SecurityBundle/Security/FirewallMap.php

```diff
@@ -65,14 +69,62 @@ public function getFirewallConfig(Request $request)
 
     private function getFirewallContext(Request $request)
     {
-        if ($this->contexts->contains($request)) {
-            return $this->contexts[$request];
+        if ($this->contextCache->contains($request)) {
+            return $this->contextCache[$request];
         }
 
-        foreach ($this->map as $contextId => $requestMatcher) {
+        foreach ($this->contextMap as $contextId => $requestMatcher) {
             if (null === $requestMatcher || $requestMatcher->matches($request)) {
-                return $this->contexts[$request] = $this->container->get($contextId);
+                return $this->contextCache[$request] = $this->container->get($contextId);
             }
         }
     }
+
+    /**
+     * @internal
+     */
+    public function __get($name)
+    {
+        if ('map' === $name) {
+            @trigger_error(sprintf('Using the "%1$s::$map" property is deprecated since version 3.3 as it will be removed in 4.0. Use "%1$s::$contextMap" instead.', __CLASS__), E_USER_DEPRECATED);
```

Comment:
> should we do an $this->map = iterator_to_array($this->contextMap, false) when appropriate?

👍  though you say `$use_keys = false` but keys are important here, done

> also: contextMap is private, so it can't be used "instead". any other suggestions?

nope, fixed